.github/workflows/release.yml

```diff
@@ -101,4 +102,29 @@ jobs:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_CHANGELOG_WEBHOOK_URL }}
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
 
+  # We're running the CI workflow (where node v20 modules are cached) in
+  # merge_group and not on main, we need to explicitly cache node_modules here so
+  # that follow-on branches can use the cached version of node_modules rather
+  # than recreating them every time.
+  cache-for-alternate-node-versions:
+    name: Cache for Alternate Node Versions
+    runs-on: ${{ vars.RUNNER_NORMAL }}
+    timeout-minutes: ${{ fromJSON(vars.TIMEOUT_MINUTES_NORMAL) }}
+    continue-on-error: true
+
+    strategy:
+      matrix:
+        version: [ 20 ] # NOTE: 18 is cached in the main release workflow
 
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          show-progress: false
+
+      - name: Cache node_modules (Node v${{ matrix.version }})
+        uses: ./.github/actions/init
+        with:
+          node-version: ${{ matrix.version }}
+          turbo-team: ''
+          turbo-token: ''
```

Comment:
It feels a bit odd to tack this caching thing onto the release workflow, do you think it makes sense to have a standalone workflow that runs on push to `main` that makes sure the cache is setup how we expect?

Reply:
@BRKalow I was just trying to avoid having a sort of dangling workflow running in each push to `main` or `release/v4`.